src/mon/MonClient.cc

```diff
@@ -575,7 +576,8 @@ void MonClient::_reopen_session(int rank)
   ldout(cct, 10) << __func__ << " rank " << rank << dendl;
 
   // save global_id if any before nuking active_con
-  const uint64_t global_id = active_con ? active_con->get_global_id() : 0;
+  if (active_con)
+    global_id = active_con->get_global_id();
```

Comment:
i don't understand how this fix helps by keeping `global_id` around as a member variable. as `global_id` is assigned by the monitor, and is tracked by `active_conn` after `MAuthReply` is handled. but if it has never received the `MAuthReply` from monitor, how come we have a non-zero global_id?